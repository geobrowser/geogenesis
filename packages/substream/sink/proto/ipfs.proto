syntax = "proto3";

message IpfsMetadata {
  // We version the data structured used to represent proposal metadata. Each
  // proposal type has their own metadata and versioning that we can change
  // independently of other proposal types.
  string version = 1;
  IpfsContentType type = 2;
  string id = 3;
  string name = 4;
}

enum IpfsContentType {
  UNKNOWN = 0;
  EDIT = 1;
  CREATE_SPACE = 2;
  ADD_SUBSPACE = 3;
  REMOVE_SUBSPACE = 4;
  ADD_EDITOR = 5;
  REMOVE_EDITOR = 6;
  ADD_MEMBER = 7;
  REMOVE_MEMBER = 8;
}

message Edit {
  string version = 1;
  IpfsContentType type = 2;
  string id = 3;
  string name = 4;
  repeated Op ops = 5;
  repeated string authors = 6;
}

message Op {
  OpType opType = 1;
  Payload payload = 2;
}

message Payload {
  bytes entityId = 1;
  bytes attributeId = 2;
  Value value = 3;
}

message Value {
  ValueType type = 1;
  string value = 2;
}

enum OpType {
  NONE = 0;
  SET_TRIPLE = 1;
  DELETE_TRIPLE = 2;
}

enum ValueType {
  NULL = 0;
  TEXT = 1;
  NUMBER = 2;
  ENTITY = 3;
  COLLECTION = 4;
  CHECKBOX = 5;
  URL = 6;
  TIME = 7;
  GEO_LOCATION = 8;
}

message Membership {
  string version = 1;
  IpfsContentType type = 2;
  string id = 3;
  string name = 4;
  string userAddress = 5;
}

message Subspace {
  string version = 1;
  IpfsContentType type = 2;
  string id = 3;
  string name = 4;
  string subspace = 5;
}

message Import {
  string previousNetwork = 1;
  string previousAddress = 2;
  repeated string edits = 3;
}
