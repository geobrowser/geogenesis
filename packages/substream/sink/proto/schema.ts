import { z } from 'zod';

export const ZodIpfsMetadata = z.object({
  version: z.string().superRefine(v => v === '1.0.0'),
  type: z.union([z.literal('ADD_EDIT'), z.literal('IMPORT_SPACE')]),
});

const ZodValueOptions = z
  .object({
    language: z.optional(z.string()),
    unit: z.optional(z.string()),
    format: z.optional(z.string()),
  })
  .optional();

const ZodEditSetTriplePayload = z.object({
  entity: z.string(),
  attribute: z.string(),
  // zod has issues with discriminated unions. We set the value
  // to any here and trust that it is constructed into the correct
  // format once it's decoded.
  value: z.object({
    options: ZodValueOptions,
    value: z.string(),
    type: z.number().transform(t => {
      switch (t) {
        case 1:
          return 'TEXT';
        case 2:
          return 'NUMBER';
        case 3:
          return 'CHECKBOX';
        case 4:
          return 'URL';
        case 5:
          return 'TIME';
        case 6:
          return 'POINT';
        default:
          return 'TEXT';
      }
    }),
  }),
});

const ZodEditDeleteTriplePayload = z.object({
  entity: z.string(),
  attribute: z.string(),
});

const ZodEditSetTripleOp = z.object({
  type: z
    .literal(1)
    .transform(() => 'SET_TRIPLE')
    .superRefine(arg => arg === 'SET_TRIPLE'),
  triple: ZodEditSetTriplePayload,
});

const ZodEditDeleteTripleOp = z.object({
  type: z
    .literal(2)
    .transform(() => 'DELETE_TRIPLE')
    .superRefine(arg => arg === 'DELETE_TRIPLE'),
  triple: ZodEditDeleteTriplePayload,
});

const ZodCreateRelationOp = z.object({
  type: z
    .literal(5)
    .transform(() => 'CREATE_RELATION')
    .superRefine(arg => arg === 'CREATE_RELATION'),
  relation: z.object({
    id: z.string(),
    index: z.string(),
    fromEntity: z.string(),
    toEntity: z.string(),
    type: z.string(),
  }),
});

const ZodDeleteRelationOp = z.object({
  type: z
    .literal(6)
    .transform(() => 'DELETE_RELATION')
    .superRefine(arg => arg === 'DELETE_RELATION'),
  relation: z.object({
    id: z.string(),
  }),
});

const ZodCsvMetadata = z.object({
  type: z.literal('CSV'),
  columns: z.array(
    z.object({
      id: z.string(),
      // @TODO: Is this a number or a string?
      type: z.number().transform(t => {
        switch (t) {
          case 1:
            return 'TEXT';
          case 2:
            return 'NUMBER';
          case 3:
            return 'CHECKBOX';
          case 4:
            return 'URL';
          case 5:
            return 'TIME';
          case 6:
            return 'POINT';
          case 7:
            return 'RELATION';
          default:
            return 'TEXT';
        }
      }),
      options: ZodValueOptions,
    })
  ),
});

const ZodImportFileOp = z.object({
  type: z
    .literal(7)
    .transform(() => 'IMPORT_FILE')
    .superRefine(arg => arg === 'IMPORT_FILE'),
  url: z.string(),
  metadata: ZodCsvMetadata,
});

export const ZodOp = z.union([
  ZodEditSetTripleOp,
  ZodEditDeleteTripleOp,
  ZodCreateRelationOp,
  ZodDeleteRelationOp,
  ZodImportFileOp,
]);
export type ZodOp = z.infer<typeof ZodOp>;

export const ZodEdit = z.object({
  version: z.string(),
  type: z
    .literal(1)
    .transform(() => 'ADD_EDIT')
    .superRefine(arg => arg === 'ADD_EDIT'),
  id: z.string(),
  name: z.string(),
  ops: z.array(ZodOp),
  authors: z.array(z.string()),
});

export type DecodedEdit = z.infer<typeof ZodEdit>;

export const ZodImportEdit = ZodEdit.extend({
  createdBy: z.string(),
  createdAt: z.string(),
});

export type DecodedImportEdit = z.infer<typeof ZodImportEdit>;
