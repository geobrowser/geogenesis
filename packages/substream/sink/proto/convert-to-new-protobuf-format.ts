import { createGeoId } from '@geogenesis/sdk';
import fs from 'fs';

import { Edit, EditLegacy, IpfsContentType, IpfsMetadata, Membership, Subspace } from '../proto';

const write = (type: IpfsContentType, name: string, buffer: Buffer) => {
  const legacy = EditLegacy.fromBinary(buffer);

  console.log('legacy', JSON.stringify(legacy, null, 2));

  const metadata = new Edit({
    name: name,
    type: type,
    version: '1.0.0',
    proposalId: createGeoId(),
    authors: legacy.authors,
    ops: legacy.ops,
  });

  console.log('metadata', JSON.stringify(metadata.toJson(), null, 2));

  const binary = metadata.toBinary();
  fs.writeFileSync(`${name.split(' ').join('-').toLowerCase()}.pb`, binary);
};

const read = (buffer: Buffer) => {
  const data = Edit.fromBinary(buffer);
  console.log('read data', JSON.stringify(data.ops, null, 2));
  fs.writeFileSync('ops-1.json', JSON.stringify(data.ops));
};

// write(IpfsContentType.EDIT, 'Edits 1 with new schema', fs.readFileSync('edits_1.pb'));
read(fs.readFileSync('edits-1-with-new-schema.pb'));
