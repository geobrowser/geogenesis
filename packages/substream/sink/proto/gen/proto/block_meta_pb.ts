// @generated by protoc-gen-es v1.9.0 with parameter "target=ts"
// @generated from file proto/block_meta.proto (package sf.ethereum.block_meta.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck

import type { BinaryReadOptions, FieldList, JsonReadOptions, JsonValue, PartialMessage, PlainMessage } from "@bufbuild/protobuf";
import { Message, proto3, protoInt64 } from "@bufbuild/protobuf";

/**
 * @generated from message sf.ethereum.block_meta.v1.BlockMeta
 */
export class BlockMeta extends Message<BlockMeta> {
  /**
   * @generated from field: string hash = 1;
   */
  hash = "";

  /**
   * @generated from field: uint64 number = 2;
   */
  number = protoInt64.zero;

  /**
   * @generated from field: string parent_hash = 3;
   */
  parentHash = "";

  /**
   * @generated from field: string timestamp = 4;
   */
  timestamp = "";

  constructor(data?: PartialMessage<BlockMeta>) {
    super();
    proto3.util.initPartial(data, this);
  }

  static readonly runtime: typeof proto3 = proto3;
  static readonly typeName = "sf.ethereum.block_meta.v1.BlockMeta";
  static readonly fields: FieldList = proto3.util.newFieldList(() => [
    { no: 1, name: "hash", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "number", kind: "scalar", T: 4 /* ScalarType.UINT64 */ },
    { no: 3, name: "parent_hash", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 4, name: "timestamp", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ]);

  static fromBinary(bytes: Uint8Array, options?: Partial<BinaryReadOptions>): BlockMeta {
    return new BlockMeta().fromBinary(bytes, options);
  }

  static fromJson(jsonValue: JsonValue, options?: Partial<JsonReadOptions>): BlockMeta {
    return new BlockMeta().fromJson(jsonValue, options);
  }

  static fromJsonString(jsonString: string, options?: Partial<JsonReadOptions>): BlockMeta {
    return new BlockMeta().fromJsonString(jsonString, options);
  }

  static equals(a: BlockMeta | PlainMessage<BlockMeta> | undefined, b: BlockMeta | PlainMessage<BlockMeta> | undefined): boolean {
    return proto3.util.equals(BlockMeta, a, b);
  }
}

