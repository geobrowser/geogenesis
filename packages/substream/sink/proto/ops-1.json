[
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "30314f95b10d4d08972552306de3b677",
      "attributeId": "c1f4cb6fece44c3ca447ab005b756972",
      "value": { "type": "ENTITY", "value": "e9b2ca5149f743b7ab1e57f5a65d8009" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "f1b9fd886388436e95b551aafaea77e5",
      "attributeId": "8f151ba4de204e3c9cb499ddf96f48f1",
      "value": { "type": "ENTITY", "value": "1d5d0c2adb23466ca0b09abe879df457" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "f1b9fd886388436e95b551aafaea77e5",
      "attributeId": "beaba5cba67741a8b35377030613fc70",
      "value": { "type": "COLLECTION", "value": "7e3178e1e50f4c5ebc783889377a2309" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "f1b9fd886388436e95b551aafaea77e5",
      "attributeId": "34f535072e6b42c5a84443981a77cfa2",
      "value": { "type": "ENTITY", "value": "43f688823e3f4fff82d15f2ebafab79e" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "7e3178e1e50f4c5ebc783889377a2309",
      "attributeId": "8f151ba4de204e3c9cb499ddf96f48f1",
      "value": { "type": "ENTITY", "value": "c373a33052df47b3a6d2df552bda4b44" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "30314f95b10d4d08972552306de3b677",
      "attributeId": "c43b537bcff742718822717fdf2c9c01",
      "value": { "type": "ENTITY", "value": "7e3178e1e50f4c5ebc783889377a2309" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "f1b9fd886388436e95b551aafaea77e5",
      "attributeId": "9b1f76ff9711404c861e59dc3fa7d037",
      "value": { "type": "TEXT", "value": "The Root space is the top level space for the canonical global graph." }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "f1b9fd886388436e95b551aafaea77e5",
      "attributeId": "a126ca530c8e48d5b88882c734c38935",
      "value": { "type": "TEXT", "value": "Root" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "e9b2ca5149f743b7ab1e57f5a65d8009",
      "attributeId": "a126ca530c8e48d5b88882c734c38935",
      "value": { "type": "TEXT", "value": "The Root space is th" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "e9b2ca5149f743b7ab1e57f5a65d8009",
      "attributeId": "dd4999b977f04c2ba02b5a26b233854e",
      "value": { "type": "ENTITY", "value": "f1b9fd886388436e95b551aafaea77e5" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "e9b2ca5149f743b7ab1e57f5a65d8009",
      "attributeId": "f88047cebd8d4fbf83f658e84ee533e4",
      "value": {
        "type": "TEXT",
        "value": "The Root space is the top level space for the canonical global graph. Its subspaces are the top level spaces that people see on the home page. It also defines Types, Attributes, Goals, and Policies that are global in nature and can be used by other spaces.\n\n"
      }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "30314f95b10d4d08972552306de3b677",
      "attributeId": "ede47e6930b044998ea4aafbda449609",
      "value": { "type": "TEXT", "value": "a0" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "43f688823e3f4fff82d15f2ebafab79e",
      "attributeId": "8f151ba4de204e3c9cb499ddf96f48f1",
      "value": { "type": "ENTITY", "value": "11618d1215d749a181abf03447d72e86" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "30314f95b10d4d08972552306de3b677",
      "attributeId": "8f151ba4de204e3c9cb499ddf96f48f1",
      "value": { "type": "ENTITY", "value": "c167ef23fb2a40449ed945123ce7d2a9" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "e9b2ca5149f743b7ab1e57f5a65d8009",
      "attributeId": "8f151ba4de204e3c9cb499ddf96f48f1",
      "value": { "type": "ENTITY", "value": "8426caa143d647d4a6f100c7c1a9a320" }
    }
  },
  {
    "opType": "SET_TRIPLE",
    "payload": {
      "entityId": "43f688823e3f4fff82d15f2ebafab79e",
      "attributeId": "334b8ac01be14079b1707e11d0f9eb8d",
      "value": {
        "type": "TEXT",
        "value": "https://api.thegraph.com/ipfs/api/v0/cat?arg=QmX8tHpruktURDhcZ7LyN3fL9dRpuwBdv3NzSjxAzDfeX8"
      }
    }
  }
]
