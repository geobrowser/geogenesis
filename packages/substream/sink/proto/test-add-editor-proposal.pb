

Add editor0.0.10x1234"-1(