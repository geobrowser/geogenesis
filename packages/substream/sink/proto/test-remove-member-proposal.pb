
Remove member0.0.10x1234"-1(