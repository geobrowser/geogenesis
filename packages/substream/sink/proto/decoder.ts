import { Edit, Membership, Subspace } from '@geogenesis/sdk/proto';
import { Effect, Either } from 'effect';

import {
  ZodEdit,
  ZodEditorshipProposal,
  ZodMembershipProposal,
  ZodSubspaceProposal,
} from '../events/proposals-created/parser';
import { slog } from '~/sink/utils/slog';

export class CouldNotDecodeProtobufError extends Error {
  _tag: 'CouldNotDecodeProtobufError' = 'CouldNotDecodeProtobufError';
}

/**
 * decode(() => Edit.fromBinary(Buffer.from('')));
 */
export function decode<T>(fn: () => T) {
  return Effect.gen(function* (_) {
    // const telemetry = yield* _(Telemetry);

    const edit = yield* _(
      Effect.try({
        try: () => fn(),
        catch: error => new CouldNotDecodeProtobufError(String(error)),
      }),
      Effect.either
    );

    return Either.match(edit, {
      onLeft: error => {
        // telemetry.captureException(error);

        slog({
          level: 'error',
          requestId: '-1',
          message: `Could not decode protobuf
            Cause: ${error.cause}
            Message: ${error.message}
          `,
        });

        return null;
      },
      onRight: value => {
        return value;
      },
    });
  });
}

function decodeEdit(data: Buffer) {
  return Effect.gen(function* (_) {
    const decodeEffect = decode(() => {
      const edit = Edit.fromBinary(data);
      const parseResult = ZodEdit.safeParse(edit.toJson());

      if (parseResult.success) {
        return parseResult.data;
      }

      return null;
    });

    return yield* _(decodeEffect);
  });
}

function decodeMembership(data: Buffer) {
  return Effect.gen(function* (_) {
    const decodeEffect = decode(() => {
      const memberRequest = Membership.fromBinary(data);
      const parseResult = ZodMembershipProposal.safeParse(memberRequest.toJson());

      if (parseResult.success) {
        return parseResult.data;
      }

      return null;
    });

    return yield* _(decodeEffect);
  });
}

function decodeEditorship(data: Buffer) {
  return Effect.gen(function* (_) {
    const decodeEffect = decode(() => {
      const memberRequest = Membership.fromBinary(data);
      const parseResult = ZodEditorshipProposal.safeParse(memberRequest.toJson());

      if (parseResult.success) {
        return parseResult.data;
      }

      return null;
    });

    return yield* _(decodeEffect);
  });
}

function decodeSubspace(data: Buffer) {
  return Effect.gen(function* (_) {
    const decodeEffect = decode(() => {
      const subspaceRequest = Subspace.fromBinary(data);
      const parseResult = ZodSubspaceProposal.safeParse(subspaceRequest.toJson());

      if (parseResult.success) {
        return parseResult.data;
      }

      return null;
    });

    return yield* _(decodeEffect);
  });
}

export const Decoder = {
  decodeEdit,
  decodeMembership,
  decodeEditorship,
  decodeSubspace,
};
