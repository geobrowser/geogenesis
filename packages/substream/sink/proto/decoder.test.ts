import { Effect } from 'effect';
import fs from 'fs';
import { describe, it } from 'vitest';

import { Edit, IpfsContentType, IpfsMetadata, Membership, Subspace } from '../proto';
import { decode } from './decoder';

describe('decode IpfsContent', () => {
  it('decodes parsed IpfsMetadata protobuf correctly', () => {
    const fileContents = fs.readFileSync('./sink/proto/test-edit-proposal.pb');
    const result = Effect.runSync(decode(() => IpfsMetadata.fromBinary(fileContents)));

    expect(result).to.not.be.null;
    expect(result).toMatchSnapshot();
  });
});

describe('decode Edit', () => {
  it('decodes parsed EDIT protobuf correctly', () => {
    const fileContents = fs.readFileSync('./sink/proto/test-edit-proposal.pb');
    const result = Effect.runSync(decode(() => Edit.fromBinary(fileContents)));

    expect(result).to.not.be.null;
    expect(result?.type).toBe(IpfsContentType.EDIT);
    expect(result?.name).toBe('Edit');
    expect(result).toMatchSnapshot();
  });
});

describe('decode Membership', () => {
  it('decodes parsed ADD_EDITOR protobuf correctly', () => {
    const fileContents = fs.readFileSync('./sink/proto/test-add-editor-proposal.pb');
    const result = Effect.runSync(decode(() => Membership.fromBinary(fileContents)));

    expect(result).to.not.be.null;
    expect(result?.type).toBe(IpfsContentType.ADD_EDITOR);
    expect(result?.name).toBe('Add editor');
    expect(result?.userAddress).toBe('0x1234');
    expect(result).toMatchSnapshot();
  });

  it('decodes parsed REMOVE_EDITOR protobuf correctly', () => {
    const fileContents = fs.readFileSync('./sink/proto/test-remove-editor-proposal.pb');
    const result = Effect.runSync(decode(() => Membership.fromBinary(fileContents)));

    expect(result).to.not.be.null;
    expect(result?.type).toBe(IpfsContentType.REMOVE_EDITOR);
    expect(result?.name).toBe('Remove editor');
    expect(result?.userAddress).toBe('0x1234');
    expect(result).toMatchSnapshot();
  });

  it('decodes parsed ADD_MEMBER protobuf correctly', () => {
    const fileContents = fs.readFileSync('./sink/proto/test-add-member-proposal.pb');
    const result = Effect.runSync(decode(() => Membership.fromBinary(fileContents)));

    expect(result).to.not.be.null;
    expect(result?.type).toBe(IpfsContentType.ADD_MEMBER);
    expect(result?.name).toBe('Add member');
    expect(result?.userAddress).toBe('0x1234');
    expect(result).toMatchSnapshot();
  });

  it('decodes parsed REMOVE_MEMBER protobuf correctly', () => {
    const fileContents = fs.readFileSync('./sink/proto/test-remove-member-proposal.pb');
    const result = Effect.runSync(decode(() => Membership.fromBinary(fileContents)));

    expect(result).to.not.be.null;
    expect(result?.type).toBe(IpfsContentType.REMOVE_MEMBER);
    expect(result?.name).toBe('Remove member');
    expect(result?.userAddress).toBe('0x1234');
    expect(result).toMatchSnapshot();
  });
});

describe('decode Subspace', () => {
  it('decodes parsed ADD_SUBSPACE protobuf correctly', () => {
    const fileContents = fs.readFileSync('./sink/proto/test-add-subspace-proposal.pb');
    const result = Effect.runSync(decode(() => Subspace.fromBinary(fileContents)));

    expect(result).to.not.be.null;
    expect(result?.type).toBe(IpfsContentType.ADD_SUBSPACE);
    expect(result?.name).toBe('Add subspace');
    expect(result?.subspace).toBe('0x1234');
    expect(result).toMatchSnapshot();
  });

  it('decodes parsed REMOVE_SUBSPACE protobuf correctly', () => {
    const fileContents = fs.readFileSync('./sink/proto/test-remove-subspace-proposal.pb');
    const result = Effect.runSync(decode(() => Subspace.fromBinary(fileContents)));

    expect(result).to.not.be.null;
    expect(result?.type).toBe(IpfsContentType.REMOVE_SUBSPACE);
    expect(result?.name).toBe('Remove subspace');
    expect(result?.subspace).toBe('0x1234');
    expect(result).toMatchSnapshot();
  });
});
