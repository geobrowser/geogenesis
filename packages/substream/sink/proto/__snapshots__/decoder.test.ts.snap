// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`decode Edit > decodes parsed EDIT protobuf correctly 1`] = `
{
  "authors": [],
  "name": "Edit",
  "ops": [],
  "proposalId": "-1",
  "type": "EDIT",
  "version": "0.0.1",
}
`;

exports[`decode IpfsContent > decodes parsed IpfsMetadata protobuf correctly 1`] = `
{
  "name": "Edit",
  "proposalId": "-1",
  "type": "EDIT",
  "version": "0.0.1",
}
`;

exports[`decode Membership > decodes parsed ADD_EDITOR protobuf correctly 1`] = `
{
  "name": "Add editor",
  "proposalId": "-1",
  "type": "ADD_EDITOR",
  "userAddress": "0x1234",
  "version": "0.0.1",
}
`;

exports[`decode Membership > decodes parsed ADD_MEMBER protobuf correctly 1`] = `
{
  "name": "Add member",
  "proposalId": "-1",
  "type": "ADD_MEMBER",
  "userAddress": "0x1234",
  "version": "0.0.1",
}
`;

exports[`decode Membership > decodes parsed REMOVE_EDITOR protobuf correctly 1`] = `
{
  "name": "Remove editor",
  "proposalId": "-1",
  "type": "REMOVE_EDITOR",
  "userAddress": "0x1234",
  "version": "0.0.1",
}
`;

exports[`decode Membership > decodes parsed REMOVE_MEMBER protobuf correctly 1`] = `
{
  "name": "Remove member",
  "proposalId": "-1",
  "type": "REMOVE_MEMBER",
  "userAddress": "0x1234",
  "version": "0.0.1",
}
`;

exports[`decode Subspace > decodes parsed ADD_SUBSPACE protobuf correctly 1`] = `
{
  "name": "Add subspace",
  "proposalId": "-1",
  "subspace": "0x1234",
  "type": "ADD_SUBSPACE",
  "version": "0.0.1",
}
`;

exports[`decode Subspace > decodes parsed ADD_SUBSPACE protobuf correctly 2`] = `
{
  "name": "Add subspace",
  "proposalId": "-1",
  "subspace": "0x1234",
  "type": "ADD_SUBSPACE",
  "version": "0.0.1",
}
`;

exports[`decode Subspace > decodes parsed REMOVE_SUBSPACE protobuf correctly 1`] = `
{
  "name": "Remove subspace",
  "proposalId": "-1",
  "subspace": "0x1234",
  "type": "REMOVE_SUBSPACE",
  "version": "0.0.1",
}
`;
